EIPS/eip-5289.md

```diff
@@ -0,0 +1,74 @@
+---
+eip: 5289
+title: Ethereum Notary Interface
+description: Allows Smart Contracts to be Legally Binding Off-Chain
+author: Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/pr-5289-discussion-notary-interface/9980
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-16
+requires: 165
+---
+
+## Abstract
+
+Currently, the real-world applications of smart contracts are limited by the fact that they aren't legally binding. This EIP proposes a standard that allows smart contracts to be legally binding by providing IPFS links to legal documents and ensuring that the users of the smart contract have privity with the relevant legal documents.
+
+## Motivation
+
+NFTs have oftentimes been branded as a way to hold and prove copyright of a specific work. However, this, in practice, has almost never been the case. Most of the time, NFTs have no legally-binding meaning, and in the rare cases that do, the NFT simply provides a limited license for the initial holder to use the work (but cannot provide any license for any future holders).
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Legal Contract Library Interface
+
+See [`IERC5289Library`](../assets/eip-5289/interfaces/IERC5289Library.sol).
+
+### Requesting a Signature
+
+To request that certain documents be signed, revert with the following reason:
+
+```solidity
+string.concat("5289:", libraryAddress1, "-", documentId1OfAddress1, "-", documentId2OfAddress1 ",", libraryAddress2, "-", documentId2, ...)
+```
+
+NOTE: If an address begins with one or more zeros, they may be omitted.
+NOTE 2: The document IDs are represented in hexadecimal.
+
+Example:
+
+```solidity
+"5289:0x1-1-2,0xdead-7b3"
+```
+
+### Signing a Document
+
+When a signature is requested, wallets MUST call `legalDocument` and fetch the file off of IPFS, and render that file to the user. If the user agrees, the wallet MUST call `signDocument`. Using a form of account abstraction is RECOMMENDED.
+
+## Rationale
+
+- `uint64` was chosen for the timestamp return type as 64-bit time registers are standard.
+- `uint16` was chosen for the document ID as 65536 documents are likely sufficient for any use case, and the contract can always be re-deployed.
+- `signDocument` used to take a signature, but due to account abstraction being imminent, this was deemed unnecessary.
+- IPFS is mandatory because the authenticity of the signed document can be proven.
+
+## Backwards Compatibility
+
+No backwards compatibility issues found.
+
+## Reference Implementation
+
+### Legal Contract Library
+
+See [`ERC5289Library`](../assets/eip-5289/ERC5289Library.sol).
+
+## Security Considerations
+
+No security considerations.
```

Comment:
Not required for a draft, but a note about the definition of privity and legal jurisdictions might be nice before final